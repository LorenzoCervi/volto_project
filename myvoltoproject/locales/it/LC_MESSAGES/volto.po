msgid ""
msgstr ""
"Project-Id-Version: \n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: \n"
"PO-Revision-Date: \n"
"Last-Translator: \n"
"Language: \n"
"Language-Team: \n"
"Content-Type: \n"
"Content-Transfer-Encoding: \n"
"Plural-Forms: \n"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Age"
msgstr "Età"

#: components/Blocks/MyComp/MySelectSidebar
msgid "City"
msgstr "Città"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Email"
msgstr "Email"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Name"
msgstr "Nome"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Notes"
msgstr "Note"

#: components/Blocks/MyComp/View
msgid "Send"
msgstr "INVIA"

#: components/Blocks/MyComp/MySelectSidebar
msgid "Sex"
msgstr "Sesso"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "age"
msgstr "Età"

#: components/Blocks/MyComp/View
msgid "assistent"
msgstr "Parla con un Esperto"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "city"
msgstr "Città"

#: components/Blocks/MyComp/View
msgid "errAfield"
msgstr "Errore in tutti i campi"

#: components/Blocks/MyComp/View
msgid "errAge"
msgstr "Errore contenuto Età"

#: components/Blocks/MyComp/View
msgid "errCity"
msgstr "Errore contenuto Città"

#: components/Blocks/MyComp/View
msgid "errEmail"
msgstr "Errore contenuto Email"

#: components/Blocks/MyComp/View
msgid "errName"
msgstr "Errore contenuto Nome"

#: components/Blocks/MyComp/View
msgid "errNotes"
msgstr "Errore contenuto Note"

#: components/Blocks/MyComp/View
msgid "errSex"
msgstr "Errore contenuto Sesso"

#: components/Blocks/MyComp/View
msgid "errcnt1"
msgstr "Questo input è sbagliato, leggere l'errore nel campo"

#: components/Blocks/MyComp/View
msgid "errcnt2"
msgstr "Questo campo è obbligatorio"

#: components/Blocks/MyComp/View
msgid "errcntA"
msgstr "Nessun campo richiesto è riempito,tranne Età e Motivo(default)"

#: components/Blocks/MyComp/View
msgid "female"
msgstr "Femmina"

#: components/Blocks/MyComp/View
msgid "help"
msgstr "Aiuto"

#: components/Blocks/MyComp/View
msgid "invAge"
msgstr "Età in input SBAGLIATA (range:18-100)"

#: components/Blocks/MyComp/View
msgid "invCity"
msgstr "Città in input SBAGLIATA,(deve essere maiuscola) e deve essere almeno lungo 2 lettere"

#: components/Blocks/MyComp/View
msgid "invEmail"
msgstr "Email in input SBAGLIATA (miaemail@dominio.nat)"

#: components/Blocks/MyComp/View
msgid "invName"
msgstr "Nome NON PUÒ contenere numeri o simboli e deve essere almeno lungo 2 lettere"

#: components/Blocks/MyComp/View
msgid "job"
msgstr "Richiesta lavoro"

#: components/Blocks/MyComp/View
msgid "male"
msgstr "Maschio"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "name"
msgstr "Nome"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "notes"
msgstr "Note"

#: components/Blocks/MyComp/View
msgid "other"
msgstr "Altro"

#: components/Blocks/MyComp/MySelectSidebar
msgid "placeholder"
msgstr "Suggerimento"

#: components/Blocks/MyComp/View
msgid "question"
msgstr "Domanda"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "reason"
msgstr "Motivo"

#: components/Blocks/MyComp/MySelectSidebar
msgid "required"
msgstr "Campo obbligatorio"

#: components/Blocks/MyComp/Edit
#: components/Blocks/MyComp/View
msgid "sex"
msgstr "Sesso"

#: components/Blocks/MyComp/View
msgid "sexdef"
msgstr "Scegli"

#: components/Blocks/MyComp/View
msgid "succ"
msgstr "Successo!"

#: components/Blocks/MyComp/View
msgid "succSub"
msgstr "Tutti i campi sono stati inviati"
