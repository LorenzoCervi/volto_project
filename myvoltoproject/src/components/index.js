/**
 * Add your components here.
 * @module components
 * @example
 * import Footer from './Footer/Footer';
 *
 * export {
 *   Footer,
 * };
 */
import EmailWidget from './EmailWidget/EmailWidget';
import NumberWidget from './NumberWidget/NumberWidget';

import SelectMyWidget from './SelectMyWidget/SelectMyWidget';

import MyFormView from './Blocks/MyComp/View';
import MyFormEdit from './Blocks/MyComp/Edit';

export { MyFormView, MyFormEdit, EmailWidget, NumberWidget, SelectMyWidget };
