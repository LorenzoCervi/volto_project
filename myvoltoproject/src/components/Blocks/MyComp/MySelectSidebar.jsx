import React from 'react';
import { Segment } from 'semantic-ui-react';
import { CheckboxWidget, TextWidget } from '@plone/volto/components';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';

const messages = defineMessages({
  Placeholder: {
    id: 'placeholder',
    defaultMessage: 'Placeholder',
  },
  Required: {
    id: 'required',
    defaultMessage: 'Required',
  },
});

const MySelectSidebar = (props) => {

  return (
    <Segment.Group raised>
      <header>
        <h2>
          <FormattedMessage id="Name" defaultMessage="Name" />
        </h2>
      </header>
      <Segment className="form sidebar-name-plch">
        <TextWidget
          id="placeholder-name"
          title={props.intl.formatMessage(messages.Placeholder)}
          value={props.data?.placeholderN ?? ''}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              placeholderN: v,
            });
          }}
        />
      </Segment>
      <Segment className="form sidebar-name-req">
        <CheckboxWidget
          id="required-name"
          title={props.intl.formatMessage(messages.Required)}
          value={props.data?.requiredName ?? false}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              requiredName: v,
            });
          }}
        />
      </Segment>

      <header>
        <h2>
          <FormattedMessage id="Age" defaultMessage="Age" />
        </h2>
      </header>
      <Segment className="form sidebar-age-req">
        <CheckboxWidget
          id="required-age"
          title={props.intl.formatMessage(messages.Required)}
          value={props.data?.requiredAge ?? false}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              requiredAge: v,
            });
          }}
        />
      </Segment>

      <header>
        <h2>
          <FormattedMessage id="Sex" defaultMessage="Sex" />
        </h2>
      </header>
      <Segment className="form sidebar-sex-req">
        <CheckboxWidget
          id="required-sex"
          title={props.intl.formatMessage(messages.Required)}
          value={props.data?.requiredSex ?? false}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              requiredSex: v,
            });
          }}
        />
      </Segment>

      <header>
        <h2>
          <FormattedMessage id="City" defaultMessage="City" />
        </h2>
      </header>
      <Segment className="form sidebar-city-plch">
        <TextWidget
          id="placeholder-city"
          title={props.intl.formatMessage(messages.Placeholder)}
          value={props.data?.placeholderC ?? ''}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              placeholderC: v,
            });
          }}
        />
      </Segment>
      <Segment className="form sidebar-city-req">
        <CheckboxWidget
          id="required-city"
          title={props.intl.formatMessage(messages.Required)}
          value={props.data?.requiredCity ?? false}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              requiredCity: v,
            });
          }}
        />
      </Segment>

      <header>
        <h2>
          <FormattedMessage id="Notes" defaultMessage="Notes" />
        </h2>
      </header>
      <Segment className="form sidebar-notes-plch">
        <TextWidget
          id="placeholder-notes"
          title={props.intl.formatMessage(messages.Placeholder)}
          value={props.data?.placeholderNotes ?? ''}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              placeholderNotes: v,
            });
          }}
        />
      </Segment>
      <Segment className="form sidebar-notes-req">
        <CheckboxWidget
          id="required-notes"
          title={props.intl.formatMessage(messages.Required)}
          value={props.data?.requiredNotes ?? false}
          onChange={(e, v) => {
            props.onChangeBlock(props.block, {
              ...props.data,
              requiredNotes: v,
            });
          }}
        />
      </Segment>
    </Segment.Group>
  );
};

MySelectSidebar.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
};

export default injectIntl(MySelectSidebar);
