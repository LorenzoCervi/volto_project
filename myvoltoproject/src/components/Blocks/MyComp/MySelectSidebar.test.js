import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import { CheckboxWidget, TextWidget } from '@plone/volto/components';

import MySelectSidebar from './MySelectSidebar.jsx';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});

const mycomp = (
  <Provider store={store}>
    <MySelectSidebar 
    data={{}} 
    block="BLOCK" 
    onChangeBlock={() => {}} 
    />
  </Provider>
);

//SNAPSHOT
test('render the entire component in a Snapshot using the sidebar component', () => {
  const component = renderer.create(mycomp);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

//UNIT TEST

//SUITE PER CONTARE GLI ELEMENTI NELLA SIDEBAR
describe('searching',()=>{
  test('input field textbox',()=>{
  	const {getAllByRole}=render(mycomp);
  	const input=getAllByRole('textbox',{nome:''});
  	expect(input.length).toBe(3);
  });
   test('input field checkbox',()=>{
  	const {getAllByRole}=render(mycomp);
  	const input=getAllByRole('checkbox',{nome:''});
  	expect(input.length).toBe(5);
  });
     test('heading',()=>{
  	const {getAllByRole}=render(mycomp);
  	const input=getAllByRole('heading',{nome:''});
  	expect(input.length).toBe(5);
  });
});



//SUITE PER LE PROPS
//PER TEXTBOX
const {getAllByRole}=render(mycomp);
describe('texboxes props', () => {
	const textbox=getAllByRole('textbox',{nome:''});
	const Name=textbox[0];
	const City=textbox[1];
	const Notes=textbox[2];
  test('name input id', () => {
  	expect(Name).toHaveProperty('id', 'field-placeholder-name'); 
  });
  test('name input name', () => {
  	expect(Name).toHaveProperty('name', 'placeholder-name'); 
  });
  test('name input value', () => {
  	expect(Name).toHaveProperty('value', ''); 
  });
  test('name input type', () => {
  	expect(Name).toHaveProperty('type', 'text'); 
  });
  test('city input id', () => {
  	expect(City).toHaveProperty('id', 'field-placeholder-city'); 
  });
  test('city input name', () => {
  	expect(City).toHaveProperty('name', 'placeholder-city'); 
  });
  test('city input value', () => {
  	expect(City).toHaveProperty('value', ''); 
  });
  test('city input type', () => {
  	expect(City).toHaveProperty('type', 'text'); 
  });
    test('notes input id', () => {
  	expect(Notes).toHaveProperty('id', 'field-placeholder-notes'); 
  });
  test('notes input name', () => {
  	expect(Notes).toHaveProperty('name', 'placeholder-notes'); 
  });
  test('notes input value', () => {
  	expect(Notes).toHaveProperty('value', ''); 
  });
  test('notes input type', () => {
  	expect(Notes).toHaveProperty('type', 'text'); 
  });
});

describe('checkboxes props', () => {
	const checkbox=getAllByRole('checkbox',{nome:''});
	const Name=checkbox[0];
	const Age=checkbox[1];
	const Sex=checkbox[2];
	const City=checkbox[3];
	const Notes=checkbox[4];
  test('name input name', () => {
  	expect(Name).toHaveProperty('name', 'field-required-name'); 
  });
  test('name input value', () => {
  	expect(Name).toHaveProperty('value', ''); 
  });
  test('age input name', () => {
  	expect(Age).toHaveProperty('name', 'field-required-age'); 
  });
  test('age input value', () => {
  	expect(Age).toHaveProperty('value', ''); 
  });
  test('sex input name', () => {
  	expect(Sex).toHaveProperty('name', 'field-required-sex'); 
  });
  test('sex input value', () => {
  	expect(Sex).toHaveProperty('value', ''); 
  });
  test('city input name', () => {
  	expect(City).toHaveProperty('name', 'field-required-city'); 
  });
  test('city input value', () => {
  	expect(City).toHaveProperty('value', ''); 
  });
  test('notes input name', () => {
  	expect(Notes).toHaveProperty('name', 'field-required-notes'); 
  });
  test('notes input value', () => {
  	expect(Notes).toHaveProperty('value', ''); 
  });
});

