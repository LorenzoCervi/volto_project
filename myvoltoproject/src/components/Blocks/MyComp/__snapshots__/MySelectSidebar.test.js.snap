// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the entire component in a Snapshot using the sidebar component 1`] = `
<div
  className="ui raised segments"
>
  <header>
    <h2>
      Name
    </h2>
  </header>
  <div
    className="ui segment form sidebar-name-plch"
  >
    <div
      className="inline field text"
      id="field-placeholder-name"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-placeholder-name"
              >
                Placeholder
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui input"
            >
              <input
                id="field-placeholder-name"
                maxLength={null}
                minLength={null}
                name="placeholder-name"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui segment form sidebar-name-req"
  >
    <div
      className="inline field"
      id="field-required-name"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-required-name"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-required-name"
                >
                  Required
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header>
    <h2>
      Age
    </h2>
  </header>
  <div
    className="ui segment form sidebar-age-req"
  >
    <div
      className="inline field"
      id="field-required-age"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-required-age"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-required-age"
                >
                  Required
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header>
    <h2>
      Sex
    </h2>
  </header>
  <div
    className="ui segment form sidebar-sex-req"
  >
    <div
      className="inline field"
      id="field-required-sex"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-required-sex"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-required-sex"
                >
                  Required
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header>
    <h2>
      City
    </h2>
  </header>
  <div
    className="ui segment form sidebar-city-plch"
  >
    <div
      className="inline field text"
      id="field-placeholder-city"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-placeholder-city"
              >
                Placeholder
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui input"
            >
              <input
                id="field-placeholder-city"
                maxLength={null}
                minLength={null}
                name="placeholder-city"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui segment form sidebar-city-req"
  >
    <div
      className="inline field"
      id="field-required-city"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-required-city"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-required-city"
                >
                  Required
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header>
    <h2>
      Notes
    </h2>
  </header>
  <div
    className="ui segment form sidebar-notes-plch"
  >
    <div
      className="inline field text"
      id="field-placeholder-notes"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="four wide column"
          >
            <div
              className="wrapper"
            >
              <label
                htmlFor="field-placeholder-notes"
              >
                Placeholder
              </label>
            </div>
          </div>
          <div
            className="eight wide column"
          >
            <div
              className="ui input"
            >
              <input
                id="field-placeholder-notes"
                maxLength={null}
                minLength={null}
                name="placeholder-notes"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ui segment form sidebar-notes-req"
  >
    <div
      className="inline field"
      id="field-required-notes"
    >
      <div
        className="ui grid"
      >
        <div
          className="stretched row"
        >
          <div
            className="twelve wide column"
          >
            <div
              className="wrapper"
            >
              <div
                className="ui checkbox"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  name="field-required-notes"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="field-required-notes"
                >
                  Required
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
